src/roller/table.ts

```diff
@@ -38,40 +43,148 @@ export class TableRoller extends GenericFileRoller<string> {
     async build() {
         this.resultEl.empty();
         const result = [this.result];
-
         if (this.plugin.data.displayResultsInline) {
             result.unshift(this.inlineText);
         }
-
         MarkdownRenderer.renderMarkdown(
             result.join(""),
             this.resultEl.createSpan("embedded-table-result"),
             this.source,
             null
         );
     }
+
+    // Quick and Dirty
+    // TODO: to be refactored (using regex maybe?)
+    prettify(input: string): string {
+        let tab = "\t";
+
+        let tabCount = 0;
+        let output:string = "";
+
+        for (let i = 0; i < input.length; i++) {
+            if (input.charAt(i) == "(") {
+                tabCount++;
+                output += "(\n";
+                output += tab.repeat(tabCount);
+            }
+            else if (input.charAt(i) == ")") {
+                tabCount--;
+                output += "\n";
+                output += tab.repeat(tabCount);
+                output += ")";
+            }
+            else if (input.charAt(i) == ";") {
+                output += ",\n";
+                output += tab.repeat(tabCount);
+            }
+            else if (input.charAt(i) == "|" && input.charAt(i-1) == "|") {
+                output += "|\n";
+                output += tab.repeat(tabCount);
+            }
+            else {
+                output += input.charAt(i);
+            }
+        }
+        return output;
+    }
+
+    async getSubResult(input: string): Promise<SubRollerResult> {
+        let res: SubRollerResult = new SubRollerResult();
+        res.result = input;
+
+        let subTooltips: string[] = [];
+
+        // WARN: we may receive an input that is not string (but a number). Check
+        // for embeded formulas only if we can.
+        if (typeof input?.matchAll === "function") {
```

Comment:
You can test directly if this is a string:

`typeof input == "string"`

Reply:
Yeah, I agree that was too much!
Will change it.